src/commands/deployment/gate.ts

```diff
@@ -255,21 +253,21 @@ export class DeploymentGateCommand extends Command {
 
             return 1
           case 'in_progress':
-            const waitTime = Math.min(this.pollingInterval, remainingTime)
-            const waitTimeInSeconds = Math.floor(waitTime / 1000)
             const rules = response.data.data.attributes.rules
             const totalRules = rules.length
             const completedRules = rules.filter((rule) => rule.status !== 'in_progress').length
 
             this.logger.info(
               `\tGate evaluation in progress (${completedRules}/${totalRules} rules completed). Retrying in ${waitTimeInSeconds}s...`
             )
+            break
 
-            await new Promise((resolve) => setTimeout(resolve, waitTime))
-            continue
           default:
-            throw new Error(`Unknown gate evaluation status: ${status}`)
+            this.logger.warn(`Unknown gate evaluation status: ${status}, retrying in ${waitTimeInSeconds}s...`)
         }
+
+        await new Promise((resolve) => setTimeout(resolve, waitTime))
+        timePassed = Date.now() - this.startTime
```

Comment:
I guess the last retry might not be done based on this? if the max wait is 10 seconds, we will do a single poll, then wait for 10 seconds and then update this value which will be slightly higher than 10 seconds, so we will exit the while?